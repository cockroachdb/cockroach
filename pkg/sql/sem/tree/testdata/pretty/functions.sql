select
	min(a,b),
	min(distinct a,b),
	min(),
	min() over (),
	min() over (order by x),
	min() filter (where x >3 and y < 4),
	min() filter (where x >3 and y < 4) over (order by x),
	min() over (range between unbounded preceding and unbounded following),
	min() over (rows between 1 following and 1 following),
	min() over (w partition by a,b order by x,y rows between 1 following and 1 following)
