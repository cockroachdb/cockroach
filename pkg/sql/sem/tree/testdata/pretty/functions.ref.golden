// Code generated by TestPretty. DO NOT EDIT.
// GENERATED FILE DO NOT EDIT
1:
-
SELECT
	min(
		a,
		b
	),
	min(
		DISTINCT
		a,
		b
	),
	min(),
	min() OVER (),
	min() OVER (
		ORDER BY
			x
	),
	min() FILTER (
		WHERE
			x
			> 3
			AND y
				< 4
	),
	min() FILTER (
		WHERE
			x
			> 3
			AND y
				< 4
	) OVER (
		ORDER BY
			x
	),
	min() OVER (
		RANGE
			BETWEEN
				UNBOUNDED PRECEDING
			AND
				UNBOUNDED FOLLOWING
	),
	min() OVER (
		ROWS
			BETWEEN
				1 FOLLOWING
			AND
				1 FOLLOWING
	),
	min() OVER (
		w
		PARTITION BY
			a,
			b
		ORDER BY
			x,
			y
		ROWS
			BETWEEN
				1 FOLLOWING
			AND
				1 FOLLOWING
	)

14:
--------------
SELECT
	min(a, b),
	min(
		DISTINCT
		a,
		b
	),
	min(),
	min() OVER (),
	min() OVER (
		ORDER BY
			x
	),
	min() FILTER (
		WHERE
			x
			> 3
			AND y
				< 4
	),
	min() FILTER (
		WHERE
			x
			> 3
			AND y
				< 4
	) OVER (
		ORDER BY
			x
	),
	min() OVER (
		RANGE
			BETWEEN
				UNBOUNDED PRECEDING
			AND
				UNBOUNDED FOLLOWING
	),
	min() OVER (
		ROWS
			BETWEEN
				1 FOLLOWING
			AND
				1 FOLLOWING
	),
	min() OVER (
		w
		PARTITION BY
			a,
			b
		ORDER BY
			x,
			y
		ROWS
			BETWEEN
				1 FOLLOWING
			AND
				1 FOLLOWING
	)

16:
----------------
SELECT
	min(a, b),
	min(
		DISTINCT
		a,
		b
	),
	min(),
	min() OVER (),
	min() OVER (
		ORDER BY
			x
	),
	min() FILTER (
		WHERE
			x
			> 3
			AND y
				< 4
	),
	min() FILTER (
		WHERE
			x
			> 3
			AND y
				< 4
	) OVER (
		ORDER BY
			x
	),
	min() OVER (
		RANGE
			BETWEEN
				UNBOUNDED PRECEDING
			AND
				UNBOUNDED FOLLOWING
	),
	min() OVER (
		ROWS
			BETWEEN
				1 FOLLOWING
			AND
				1 FOLLOWING
	),
	min() OVER (
		w
		PARTITION BY
			a, b
		ORDER BY
			x, y
		ROWS
			BETWEEN
				1 FOLLOWING
			AND
				1 FOLLOWING
	)

17:
-----------------
SELECT
	min(a, b),
	min(
		DISTINCT
		a,
		b
	),
	min(),
	min() OVER (),
	min() OVER (
		ORDER BY
			x
	),
	min() FILTER (
		WHERE
			x > 3
			AND y
				< 4
	),
	min() FILTER (
		WHERE
			x > 3
			AND y
				< 4
	) OVER (
		ORDER BY
			x
	),
	min() OVER (
		RANGE
			BETWEEN
				UNBOUNDED PRECEDING
			AND
				UNBOUNDED FOLLOWING
	),
	min() OVER (
		ROWS
			BETWEEN
				1 FOLLOWING
			AND
				1 FOLLOWING
	),
	min() OVER (
		w
		PARTITION BY
			a, b
		ORDER BY
			x, y
		ROWS
			BETWEEN
				1 FOLLOWING
			AND
				1 FOLLOWING
	)

18:
------------------
SELECT
	min(a, b),
	min(
		DISTINCT
		a,
		b
	),
	min(),
	min() OVER (),
	min() OVER (
		ORDER BY x
	),
	min() FILTER (
		WHERE
			x > 3
			AND y
				< 4
	),
	min() FILTER (
		WHERE
			x > 3
			AND y
				< 4
	) OVER (
		ORDER BY x
	),
	min() OVER (
		RANGE
			BETWEEN
				UNBOUNDED PRECEDING
			AND
				UNBOUNDED FOLLOWING
	),
	min() OVER (
		ROWS
			BETWEEN
				1 FOLLOWING
			AND
				1 FOLLOWING
	),
	min() OVER (
		w
		PARTITION BY
			a, b
		ORDER BY
			x, y
		ROWS
			BETWEEN
				1 FOLLOWING
			AND
				1 FOLLOWING
	)

21:
---------------------
SELECT
	min(a, b),
	min(
		DISTINCT
		a,
		b
	),
	min(),
	min() OVER (),
	min() OVER (
		ORDER BY x
	),
	min() FILTER (
		WHERE
			x > 3
			AND y < 4
	),
	min() FILTER (
		WHERE
			x > 3
			AND y < 4
	) OVER (
		ORDER BY x
	),
	min() OVER (
		RANGE
			BETWEEN
				UNBOUNDED PRECEDING
			AND
				UNBOUNDED FOLLOWING
	),
	min() OVER (
		ROWS
			BETWEEN
				1 FOLLOWING
			AND
				1 FOLLOWING
	),
	min() OVER (
		w
		PARTITION BY
			a, b
		ORDER BY
			x, y
		ROWS
			BETWEEN
				1 FOLLOWING
			AND
				1 FOLLOWING
	)

23:
-----------------------
SELECT
	min(a, b),
	min(DISTINCT a, b),
	min(),
	min() OVER (),
	min() OVER (
		ORDER BY x
	),
	min() FILTER (
		WHERE
			x > 3
			AND y < 4
	),
	min() FILTER (
		WHERE
			x > 3
			AND y < 4
	) OVER (
		ORDER BY x
	),
	min() OVER (
		RANGE
			BETWEEN
				UNBOUNDED PRECEDING
			AND
				UNBOUNDED FOLLOWING
	),
	min() OVER (
		ROWS
			BETWEEN
				1 FOLLOWING
			AND
				1 FOLLOWING
	),
	min() OVER (
		w
		PARTITION BY
			a, b
		ORDER BY
			x, y
		ROWS
			BETWEEN
				1 FOLLOWING
			AND
				1 FOLLOWING
	)

24:
------------------------
SELECT
	min(a, b),
	min(DISTINCT a, b),
	min(),
	min() OVER (),
	min() OVER (
		ORDER BY x
	),
	min() FILTER (
		WHERE
			x > 3
			AND y < 4
	),
	min() FILTER (
		WHERE
			x > 3
			AND y < 4
	) OVER (ORDER BY x),
	min() OVER (
		RANGE
			BETWEEN
				UNBOUNDED PRECEDING
			AND
				UNBOUNDED FOLLOWING
	),
	min() OVER (
		ROWS
			BETWEEN
				1 FOLLOWING
			AND
				1 FOLLOWING
	),
	min() OVER (
		w
		PARTITION BY
			a, b
		ORDER BY
			x, y
		ROWS
			BETWEEN
				1 FOLLOWING
			AND
				1 FOLLOWING
	)

27:
---------------------------
SELECT
	min(a, b),
	min(DISTINCT a, b),
	min(),
	min() OVER (),
	min() OVER (
		ORDER BY x
	),
	min() FILTER (
		WHERE
			x > 3 AND y < 4
	),
	min() FILTER (
		WHERE
			x > 3 AND y < 4
	) OVER (ORDER BY x),
	min() OVER (
		RANGE
			BETWEEN
				UNBOUNDED PRECEDING
			AND
				UNBOUNDED FOLLOWING
	),
	min() OVER (
		ROWS
			BETWEEN
				1 FOLLOWING
			AND
				1 FOLLOWING
	),
	min() OVER (
		w
		PARTITION BY
			a, b
		ORDER BY
			x, y
		ROWS
			BETWEEN
				1 FOLLOWING
			AND
				1 FOLLOWING
	)

28:
----------------------------
SELECT
	min(a, b),
	min(DISTINCT a, b),
	min(),
	min() OVER (),
	min() OVER (ORDER BY x),
	min() FILTER (
		WHERE
			x > 3 AND y < 4
	),
	min() FILTER (
		WHERE
			x > 3 AND y < 4
	) OVER (ORDER BY x),
	min() OVER (
		RANGE
			BETWEEN
				UNBOUNDED PRECEDING
			AND
				UNBOUNDED FOLLOWING
	),
	min() OVER (
		ROWS
			BETWEEN
				1 FOLLOWING
			AND
				1 FOLLOWING
	),
	min() OVER (
		w
		PARTITION BY
			a, b
		ORDER BY
			x, y
		ROWS
			BETWEEN
				1 FOLLOWING
			AND
				1 FOLLOWING
	)

29:
-----------------------------
SELECT
	min(a, b),
	min(DISTINCT a, b),
	min(),
	min() OVER (),
	min() OVER (ORDER BY x),
	min() FILTER (
		WHERE x > 3 AND y < 4
	),
	min() FILTER (
		WHERE x > 3 AND y < 4
	) OVER (ORDER BY x),
	min() OVER (
		RANGE
			BETWEEN
				UNBOUNDED PRECEDING
			AND
				UNBOUNDED FOLLOWING
	),
	min() OVER (
		ROWS
			BETWEEN
				1 FOLLOWING
			AND
				1 FOLLOWING
	),
	min() OVER (
		w
		PARTITION BY
			a, b
		ORDER BY
			x, y
		ROWS
			BETWEEN
				1 FOLLOWING
			AND
				1 FOLLOWING
	)

41:
-----------------------------------------
SELECT
	min(a, b),
	min(DISTINCT a, b),
	min(),
	min() OVER (),
	min() OVER (ORDER BY x),
	min() FILTER (WHERE x > 3 AND y < 4),
	min() FILTER (
		WHERE x > 3 AND y < 4
	) OVER (ORDER BY x),
	min() OVER (
		RANGE
			BETWEEN
				UNBOUNDED PRECEDING
			AND
				UNBOUNDED FOLLOWING
	),
	min() OVER (
		ROWS
			BETWEEN
				1 FOLLOWING
			AND
				1 FOLLOWING
	),
	min() OVER (
		w
		PARTITION BY
			a, b
		ORDER BY
			x, y
		ROWS
			BETWEEN
				1 FOLLOWING
			AND
				1 FOLLOWING
	)

47:
-----------------------------------------------
SELECT
	min(a, b),
	min(DISTINCT a, b),
	min(),
	min() OVER (),
	min() OVER (ORDER BY x),
	min() FILTER (WHERE x > 3 AND y < 4),
	min() FILTER (WHERE x > 3 AND y < 4) OVER (
		ORDER BY x
	),
	min() OVER (
		RANGE
			BETWEEN
				UNBOUNDED PRECEDING
			AND
				UNBOUNDED FOLLOWING
	),
	min() OVER (
		ROWS
			BETWEEN 1 FOLLOWING AND 1 FOLLOWING
	),
	min() OVER (
		w
		PARTITION BY
			a, b
		ORDER BY
			x, y
		ROWS
			BETWEEN 1 FOLLOWING AND 1 FOLLOWING
	)

48:
------------------------------------------------
SELECT
	min(a, b),
	min(DISTINCT a, b),
	min(),
	min() OVER (),
	min() OVER (ORDER BY x),
	min() FILTER (WHERE x > 3 AND y < 4),
	min() FILTER (WHERE x > 3 AND y < 4) OVER (
		ORDER BY x
	),
	min() OVER (
		RANGE
			BETWEEN
				UNBOUNDED PRECEDING
			AND
				UNBOUNDED FOLLOWING
	),
	min() OVER (
		ROWS BETWEEN 1 FOLLOWING AND 1 FOLLOWING
	),
	min() OVER (
		w
		PARTITION BY
			a, b
		ORDER BY
			x, y
		ROWS
			BETWEEN 1 FOLLOWING AND 1 FOLLOWING
	)

58:
----------------------------------------------------------
SELECT
	min(a, b),
	min(DISTINCT a, b),
	min(),
	min() OVER (),
	min() OVER (ORDER BY x),
	min() FILTER (WHERE x > 3 AND y < 4),
	min() FILTER (WHERE x > 3 AND y < 4) OVER (
		ORDER BY x
	),
	min() OVER (
		RANGE
			BETWEEN
				UNBOUNDED PRECEDING
			AND
				UNBOUNDED FOLLOWING
	),
	min() OVER (ROWS BETWEEN 1 FOLLOWING AND 1 FOLLOWING),
	min() OVER (
		w
		PARTITION BY
			a, b
		ORDER BY
			x, y
		ROWS
			BETWEEN 1 FOLLOWING AND 1 FOLLOWING
	)

59:
-----------------------------------------------------------
SELECT
	min(a, b),
	min(DISTINCT a, b),
	min(),
	min() OVER (),
	min() OVER (ORDER BY x),
	min() FILTER (WHERE x > 3 AND y < 4),
	min() FILTER (WHERE x > 3 AND y < 4) OVER (ORDER BY x),
	min() OVER (
		RANGE
			BETWEEN
				UNBOUNDED PRECEDING
			AND
				UNBOUNDED FOLLOWING
	),
	min() OVER (ROWS BETWEEN 1 FOLLOWING AND 1 FOLLOWING),
	min() OVER (
		w
		PARTITION BY
			a, b
		ORDER BY
			x, y
		ROWS
			BETWEEN 1 FOLLOWING AND 1 FOLLOWING
	)

63:
---------------------------------------------------------------
SELECT
	min(a, b),
	min(DISTINCT a, b),
	min(),
	min() OVER (),
	min() OVER (ORDER BY x),
	min() FILTER (WHERE x > 3 AND y < 4),
	min() FILTER (WHERE x > 3 AND y < 4) OVER (ORDER BY x),
	min() OVER (
		RANGE
			BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING
	),
	min() OVER (ROWS BETWEEN 1 FOLLOWING AND 1 FOLLOWING),
	min() OVER (
		w
		PARTITION BY
			a, b
		ORDER BY
			x, y
		ROWS
			BETWEEN 1 FOLLOWING AND 1 FOLLOWING
	)

65:
-----------------------------------------------------------------
SELECT
	min(a, b),
	min(DISTINCT a, b),
	min(),
	min() OVER (),
	min() OVER (ORDER BY x),
	min() FILTER (WHERE x > 3 AND y < 4),
	min() FILTER (WHERE x > 3 AND y < 4) OVER (ORDER BY x),
	min() OVER (
		RANGE BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING
	),
	min() OVER (ROWS BETWEEN 1 FOLLOWING AND 1 FOLLOWING),
	min() OVER (
		w
		PARTITION BY
			a, b
		ORDER BY
			x, y
		ROWS
			BETWEEN 1 FOLLOWING AND 1 FOLLOWING
	)

75:
---------------------------------------------------------------------------
SELECT
	min(a, b),
	min(DISTINCT a, b),
	min(),
	min() OVER (),
	min() OVER (ORDER BY x),
	min() FILTER (WHERE x > 3 AND y < 4),
	min() FILTER (WHERE x > 3 AND y < 4) OVER (ORDER BY x),
	min() OVER (RANGE BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING),
	min() OVER (ROWS BETWEEN 1 FOLLOWING AND 1 FOLLOWING),
	min() OVER (
		w
		PARTITION BY
			a, b
		ORDER BY
			x, y
		ROWS
			BETWEEN 1 FOLLOWING AND 1 FOLLOWING
	)

82:
----------------------------------------------------------------------------------
SELECT
	min(a, b),
	min(DISTINCT a, b),
	min(),
	min() OVER (),
	min() OVER (ORDER BY x),
	min() FILTER (WHERE x > 3 AND y < 4),
	min() FILTER (WHERE x > 3 AND y < 4) OVER (ORDER BY x),
	min() OVER (RANGE BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING),
	min() OVER (ROWS BETWEEN 1 FOLLOWING AND 1 FOLLOWING),
	min() OVER (
		w PARTITION BY a, b ORDER BY x, y ROWS BETWEEN 1 FOLLOWING AND 1 FOLLOWING
	)

91:
-------------------------------------------------------------------------------------------
SELECT
	min(a, b),
	min(DISTINCT a, b),
	min(),
	min() OVER (),
	min() OVER (ORDER BY x),
	min() FILTER (WHERE x > 3 AND y < 4),
	min() FILTER (WHERE x > 3 AND y < 4) OVER (ORDER BY x),
	min() OVER (RANGE BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING),
	min() OVER (ROWS BETWEEN 1 FOLLOWING AND 1 FOLLOWING),
	min() OVER (w PARTITION BY a, b ORDER BY x, y ROWS BETWEEN 1 FOLLOWING AND 1 FOLLOWING)

390:
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
SELECT
	min(a, b), min(DISTINCT a, b), min(), min() OVER (), min() OVER (ORDER BY x), min() FILTER (WHERE x > 3 AND y < 4), min() FILTER (WHERE x > 3 AND y < 4) OVER (ORDER BY x), min() OVER (RANGE BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING), min() OVER (ROWS BETWEEN 1 FOLLOWING AND 1 FOLLOWING), min() OVER (w PARTITION BY a, b ORDER BY x, y ROWS BETWEEN 1 FOLLOWING AND 1 FOLLOWING)

393:
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
SELECT min(a, b), min(DISTINCT a, b), min(), min() OVER (), min() OVER (ORDER BY x), min() FILTER (WHERE x > 3 AND y < 4), min() FILTER (WHERE x > 3 AND y < 4) OVER (ORDER BY x), min() OVER (RANGE BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING), min() OVER (ROWS BETWEEN 1 FOLLOWING AND 1 FOLLOWING), min() OVER (w PARTITION BY a, b ORDER BY x, y ROWS BETWEEN 1 FOLLOWING AND 1 FOLLOWING)


